Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -21,6 +22,12 @@ public func == (lhs: HockeyConfigData, rhs: HockeyConfigData) -> Bool {
     && lhs.userName == rhs.userName
 }
 
+public protocol NotificationAuthorizationStatusType {
+  var isAuthorized: Bool { get }
+  var isDenied: Bool { get }
+  var isNotDetermined: Bool { get }
+}
```

Comment:
Still not sure I understand why we don't use the built-in enum cases. Please elaborate why this is necessary?
And if so, can you make this an `enum` instead of a `protocol`?